server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -444,16 +444,20 @@ public CompletionStage<Result> update(
             v -> {
               DynamicForm form = formFactory.form().bindFromRequest(request);
               ImmutableMap<String, String> formData = cleanForm(form.rawData());
-
-              return applicantService.stageAndUpdateIfValid(
-                  applicantId,
-                  programId,
-                  blockId,
-                  formData,
-                  featureFlags.getFlagEnabled(
-                      request, ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ENABLED));
+              return applicantService.resetAddressCorrectionWhenAddressChanged(
```

Comment:
**thought (non-blocking):** Would it makes sense to do this inside `stageAndUpdateIfValid`?  Those functions are a bit hairy, but we might be able to reuse the roApplicantProgramService and grabbing the block.

Reply:
Unfortunately the call to reset the previously corrected address can't happen inside `stageAndUpdateIfValid`. The `stageAndUpdateIfValid` method is also called to update the address after the user selects a value from the list of address correction options, but since it is now different it would trigger the reset again and get stuck in a loop. The only context of which place the user did something is coming from knowing the controller action.